File: fds-backend/user-management-service/src/main/java/com/yokke/usermanagement/rule/RuleDto.java
package com.yokke.usermanagement.rule;

import com.yokke.base.dto.BaseDto;
import com.yokke.usermanagement.user_account_rule.UserAccountRuleDto;
import lombok.*;

import java.io.Serializable;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.stream.Collectors;

/**
 * DTO for {@link Rule}
 */
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RuleDto extends BaseDto implements Serializable {
    String ruleId;
    String ruleName;
    String ruleDescription;
    String sourceData;
    String riskLevel;
     Integer riskLevelNumber;

    List<UserAccountRuleDto> userAccountRule;

    void setUserAccountRule(List<UserAccountRuleDto> userAccountRule) {
        if (userAccountRule == null) {
            this.userAccountRule = null;
            return;
        }
        this.userAccountRule = userAccountRule.stream().sorted(
                (o1, o2) -> {
                    if (o1.getRule() == null || o2.getRule() == null) {
                        return 0;
                    }else{
                        return o1.getRule().getRiskLevelNumber().compareTo(o2.getRule().getRiskLevelNumber());
                    }
                }
        ).toList();
    }
}

File: fds-dashboard/src/packages/user-management-service/src/rule/dto/rule.dto.ts
import type { BasePageableDto } from "@/packages/base/src/dto/base-pageable.dto";
import type { BaseResponse } from "@/packages/base/src/response/base.response";
import type { UserAccountRuleDto } from "@/packages/user-management-service/src/user-account-rule/dto/user-account-rule.dto";

export type RuleDto = {
  ruleId: string;
  ruleName: string;
  ruleDescription: string;
  sourceData: string;
  riskLevel: string;
  riskLevelNumber: number;

  userAccountRule: UserAccountRuleDto[];
} & BaseResponse;

export type RulePageableDto = {
  content: RuleDto[];
} & BasePageableDto;
